core/src/main/java/discord4j/core/object/entity/channel/ThreadChannel.java

```diff
@@ -151,51 +197,160 @@ public Flux<ThreadMember> getThreadMembers(EntityRetrievalStrategy retrievalStra
+     *
+     * @param memberId The ID of the member to get permissions for.
+     * @return A {@link Mono} that, upon completion, emits a {@link PermissionSet} containing all allowed or denied
+     * permissions to the given member.
+     */
     @Override
     public Mono<PermissionSet> getEffectivePermissions(Snowflake memberId) {
         return getParent().flatMap(parent -> parent.getEffectivePermissions(memberId));
     }
 
+    /**
+     * Computes permissions given to a specified member regarding this channel.
+     *
+     * @param member The member to get permissions for.
+     * @return A {@link Mono} that, upon completion, emits a {@link PermissionSet} containing all allowed or denied
+     * permissions to the given member.
+     */
     @Override
     public Mono<PermissionSet> getEffectivePermissions(Member member) {
         return getParent().flatMap(parent -> parent.getEffectivePermissions(member));
     }
 
+    /**
+     * Tells if this thread channel supports tagging using {@link ForumTag} objects.
+     * The ability to tag a thread channel is currently related to the kind of parent channel.
+     * <br>
+     * When the channel does not support tagging, returned values from {@link ThreadChannel#getAppliedTagsIds()} and
+     * {@link ThreadChannel#getAppliedTags()} return an empty list or {@link Flux}.
+     * <br>
+     * @return A boolean indicating if the thread channel can be tagged.
+     */
+    public boolean canBeTagged() {
+        return !getData().appliedTags().isAbsent();
+    }
+
+    /**
+     * Returns the list of the applied forum tags on this thread. If the thread channel does not support tagging, the
+     * list is empty. To know if the thread supports tagging, use {@link ThreadChannel#canBeTagged()}.
+     * @see ThreadChannel#canBeTagged()
+     * @see ThreadChannel#getAppliedTagsIds()
+     *
+     * @return An {@link Optional} item, wrapping a list of applied forum or media channel tags to this thread channel.
+     * The optional item will be empty if the thread channel cannot contain tags, otherwise the list is empty.
+     * @see ThreadChannel#getAppliedTags()
+     */
+    public List<Snowflake> getAppliedTagsIds() {
+        return getData().appliedTags().toOptional()
+            .orElse(new ArrayList<>())
+            .stream().map(Snowflake::of).collect(Collectors.toList());
+    }
+
+    /**
+     * Returns the list of applied tags to this thread channel directly in the form of {@link ForumTag} objects.
+     * To only fetch the identifiers, use {@link ThreadChannel#getAppliedTagsIds()}.
+     * If the thread channel does not support tags,
+     *
+     * @return A {@link Flux}, that emits all the applied tags on this thread channel. If an error occurs, it is emitted
+     * through the flux. Only emits if the thread channel can have tags applied.
+     * @see ThreadChannel#getAppliedTagsIds()
+     */
+    public Flux<ForumTag> getAppliedTags() {
```

Comment:
for remember... in a normal channel with a thread (not forums) this class is used not? if that then this method can return a cast error?

if so maybe mention in docs about the check method and validate and throw a IlegalStateException